mono/metadata/marshal.c

```diff
@@ -1060,9 +1059,9 @@ mono_string_to_byvalstr (char *dst, MonoString *src, int size)
 		return;
 	const gsize len2 = strlen (s);
 	len = MIN (size, len2);
-	if (len >= size)
-		len--;
+	len -= (len >= size);
 	memcpy (dst, s, len);
+	dst [len] = 0;
```

Comment:
Like it says here:
https://github.com/mono/mono/pull/11985/files#diff-559b8405966c917ff45da9a2a9bf0296R1036